types/lodash/ts3.1/common/common.d.ts

```diff
@@ -0,0 +1,286 @@
+import _ = require("../index");
+// tslint:disable-next-line:strict-export-declare-modifiers
+type GlobalPartial<T> = Partial<T>;
+declare module "../index" {
+    type Omit<T, K extends keyof any> = Pick<T, Exclude<keyof T, K>>;
+    type PartialObject<T> = GlobalPartial<T>;
+    type Many<T> = T | ReadonlyArray<T>;
+    type ImpChain<T> =
+        T extends { __trapAny: any } ? Collection<any> :
+        T extends null | undefined ? Primitive<T> :
+        T extends string | null | undefined ? String :
+        T extends (...args: any) => any ? Function<T> :
+        T extends List<infer U> | null | undefined ? Collection<U> :
```

Comment:
Ah, I didn't noticate before that these chain types incorporated null/undefined already. The only issue is that when you call `value()`, the null/undefined information will be lost. Maybe that's ok?

Reply:
I think it's fine to be a bit lossy, although we might re-evaluate after people have a chance to use the new code. This only makes a difference when `first/last` happens at the end of a long chain of wrapped method calls, and I don't think that happens very often.